src/storage/aol/aol.rs

```diff
@@ -0,0 +1,354 @@
+    /// - `opts`: Configuration options for the AOL instance.
+    pub fn open(dir: &Path, opts: &Options) -> io::Result<Self> {
+        // Ensure the directory exists with proper permissions
+        Self::prepare_directory(&dir, &opts)?;
+
+        // Determine the active segment ID
+        let active_segment_id = Self::calculate_active_segment_id(&dir)?;
+
+        // Open the active segment
+        let active_segment = Segment::open(&dir, active_segment_id, &opts)?;
+
+        Ok(Self {
+            active_segment,
+            active_segment_id,
+            dir: dir.to_path_buf(),
+            opts: opts.clone(),
+            closed: false,
+            mutex: RwLock::new(()),
+        })
+    }
+
+    // Helper function to prepare the directory with proper permissions
+    fn prepare_directory(dir: &Path, opts: &Options) -> io::Result<()> {
+        fs::create_dir_all(&dir)?;
+
+        if let Ok(metadata) = fs::metadata(&dir) {
+            let mut permissions = metadata.permissions();
+            permissions.set_mode(opts.dir_mode.unwrap_or(0o750));
+            fs::set_permissions(&dir, permissions)?;
+        }
+
+        Ok(())
+    }
+
+    // Helper function to calculate the active segment ID
+    fn calculate_active_segment_id(dir: &Path) -> io::Result<u64> {
+        let (_, last) = get_segment_range(&dir)?;
+        Ok(if last > 0 { last + 1 } else { 0 })
+    }
+
+    /// Appends a record to the active segment.
+    ///
+    /// This function appends the record to the active segment. If the active segment is
+    /// full, a new segment will be created and the record will be appended to it.
+    ///
+    /// The function returns a tuple containing the offset at which the record was appended
+    /// and the number of bytes written.
+    ///
+    /// # Arguments
+    ///
+    /// * `rec` - A reference to the byte slice containing the record to be appended.
+    ///
+    /// # Returns
+    ///
+    /// A result containing the tuple `(offset, bytes_written)` or an `io::Error` in case of failure.
+    ///
+    /// # Errors
+    ///
+    /// This function may return an error if the active segment is closed, the provided record
+    /// is empty, or any I/O error occurs during the appending process.
+    pub fn append(&mut self, rec: &[u8]) -> io::Result<(u64, usize)> {
+        if self.closed {
+            return Err(io::Error::new(io::ErrorKind::Other, "Segment is closed"));
+        }
+
+        if rec.is_empty() {
+            return Err(io::Error::new(io::ErrorKind::Other, "buf is empty"));
+        }
+
+        let _lock = self.mutex.write().unwrap();
+
+        // Get options and initialize variables
+        let opts = &self.opts;
+        let mut n = 0usize;
+        let mut available = 0;
+        let mut offset = 0;
+
+        while n < rec.len() {
+            // Calculate available space in the active segment
+            available = opts.max_file_size.unwrap() - self.active_segment.offset();
```

Comment:
Is it okay to panic if there is no value in the Option? Or maybe you want to use a default value using `unwrap_or` ?

```
opts.max_file_size.unwrap_or(0)
```